apollo-federation/src/query_graph/graph_path.rs

```diff
@@ -3069,8 +3069,14 @@ impl SimultaneousPaths {
         // Pre-allocate `Vec` for the result.
         let num_options = options_for_each_path
             .iter()
-            .map(|options| options.len())
-            .product();
+            .fold(1_usize, |product, options| {
+                product.saturating_mul(options.len())
+            });
+        if num_options > 1_000_000 {
```

Comment:
this seems pretty high?

Reply:
it’s 24 MiB of memory for the Vec in the next line :shrug:

Reply:
Its just under `2^20`, which seems fine for the preview.

Reply:
That said I did pick this number out of thin air, and we can easily change it

Reply:
I think refining this number will be a topic of another meeting in a couple of weeks.